internal/plugin/otel_collector.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/nginx/agent/v3/internal/bus"
+	"github.com/nginx/agent/v3/internal/collector"
+	"github.com/nginx/agent/v3/internal/config"
+	"go.opentelemetry.io/collector/otelcol"
+)
+
+type (
+	// Collector The OTel collector plugin start an embedded OTel collector for metrics collection in the OTel format.
+	Collector struct {
+		svc     *otelcol.Collector
```

Comment:
whats `svc`?

Reply:
the running service? Collector.collector seemed incorrect. What would you prefer?

Reply:
oh right it wasn't obvious to me what `svc` meant. Could it be called just `service`?